src/test/scala/org/apache/spark/sql/delta/DeltaSymlinkManifestSuite.scala

```diff
@@ -18,35 +18,92 @@ package org.apache.spark.sql.delta
 
 import java.io.File
 import java.net.URI
+import java.util.Random
 
 import org.apache.spark.sql.delta.commands.DeltaGenerateCommand
-import org.apache.spark.sql.delta.hooks.GenerateSymlinkManifest
 import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
 import org.apache.spark.sql.delta.util.DeltaFileOperations
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs._
 import org.apache.hadoop.util.Progressable
-
 import org.apache.spark.sql._
+import org.apache.spark.sql.delta.hooks.{SymlinkGenerateManifest, JsonGenerateManifest, JsonManifest}
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.StringType
+
+class SymlinkGenerateManifestSuite
+  extends DeltaGenerateManifestSuiteBase
+    with DeltaSQLCommandTest {
+  override protected def generateSymlinkManifest(tablePath: String): Unit = {
+    val deltaLog = DeltaLog.forTable(spark, tablePath)
+    SymlinkGenerateManifest.generateFullManifest(spark, deltaLog)
+  }
+
+  override protected def getManifestLocation(): String =
+    SymlinkGenerateManifest.MANIFEST_LOCATION
+
+  override protected def mode(): String = SymlinkGenerateManifest.manifestType.mode
+
+  override protected def readManifestAsDataFilePaths(spark: SparkSession,
+                                                     manifestPath: String): Dataset[String] = {
+    import spark.implicits._
+    spark.read.text(manifestPath).select("value").as[String]
+  }
 
-class DeltaGenerateSymlinkManifestSuite
-  extends DeltaGenerateSymlinkManifestSuiteBase
-  with DeltaSQLCommandTest
+  override protected def name(): String = SymlinkGenerateManifest.name
+
+  override protected def deltaConfig(): DeltaConfig[Boolean] =
+    DeltaConfigs.SYMLINK_FORMAT_MANIFEST_ENABLED
+}
+
+class JsonGenerateManifestSuite
```

Comment:
@rahulsmahadev  The JsonGenerateManifestSuite and SymlinkGenerateManifestSuite share the same test cases except for some differences in config name, manifest path, and manifest format. Put the two test suites here